### PR TITLE
Add error page to handle fido failures

### PR DESCRIPTION
**Pull request by AfraHussaindeen**
### Purpose

- $Subject
- Sample error message[1]
- Relates to https://github.com/wso2/product-is/issues/17425

### Related PRs
- https://github.com/wso2-extensions/identity-local-auth-fido/pull/118
- https://github.com/wso2/carbon-identity-framework/pull/5118

### References 

- [1] Error page 
Screenshot 2023-11-01 at 09 06 56

> 

### Checklist
- [ ] e2e cypress tests locally verified.
- [ ] Manual test round performed and verified.
- [ ] UX/UI review done on the final implementation.
- [ ] Documentation provided. (Add links if there are any)
- [ ] Unit tests provided. (Add links if there are any)
- [ ] Integration tests provided. (Add links if there are any)

### Security checks
- [ ] Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines?
- [ ] Ran FindSecurityBugs plugin and verified report?
- [ ] Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets?

### CODE REVIEW

**Review by Achintha444 — APPROVED** [review]

File: identity-apps-core/apps/authentication-portal/src/main/webapp/fido2-error.jsp [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+<head>
+    <%-- header --%>
+    <%
+        File headerFile = new File(getServletContext().getRealPath("extensions/header.jsp"));
+        if (headerFile.exists()) {
+    %>
+    <jsp:include page="extensions/header.jsp"/>
+    <% } else { %>
+    <jsp:include page="includes/header.jsp"/>
+    <% } %>
```
Comment (pavinduLakshan) [inline identity-apps-core/apps/authentication-portal/src/main/webapp/fido2-error.jsp:69]: Let's add the analytics extension point as well, If possible.

Ref: https://github.com/wso2/identity-apps/blob/master/identity-apps-core/apps/authentication-portal/src/main/webapp/errors/error.jsp#L57-L65

File: identity-apps-core/apps/authentication-portal/src/main/webapp/fido2-error.jsp [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+</head>
+<body class="login-portal layout authentication-portal-layout">
+    <layout:main layoutName="<%= layout %>" layoutFileRelativePath="<%= layoutFileRelativePath %>" data="<%= layoutData %>" >
```
Comment (pavinduLakshan) [inline identity-apps-core/apps/authentication-portal/src/main/webapp/fido2-error.jsp:72]:
```suggestion
    <layout:main layoutName="<%= layout %>" layoutFileRelativePath="<%= layoutFileRelativePath %>" data="<%= layoutData %>">
```

File: identity-apps-core/apps/authentication-portal/src/main/webapp/fido2-error.jsp [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+        <layout:component componentName="ProductHeader">
+            <%-- product-title --%>
+            <%
+                File productTitleFile = new File(getServletContext().getRealPath("extensions/product-title.jsp"));
+                if (productTitleFile.exists()) {
+            %>
+            <jsp:include page="extensions/product-title.jsp"/>
+            <% } else { %>
+            <jsp:include page="includes/product-title.jsp"/>
```
Comment (pavinduLakshan) [inline identity-apps-core/apps/authentication-portal/src/main/webapp/fido2-error.jsp:81]: Shall we indent the statements inside if and else clauses?

File: identity-apps-core/apps/authentication-portal/src/main/webapp/fido2-error.jsp [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+<jsp:directive.include file="includes/init-url.jsp"/>
+
+<%-- Branding Preferences --%>
+<jsp:directive.include file="includes/branding-preferences.jsp"/>
+
+<%
+    session.invalidate();
+
+    String errorMessage = AuthenticationEndpointUtil.i18n(resourceBundle, "error.retry");
+    String authenticationFailed = "false";
```
Comment (pavinduLakshan) [inline identity-apps-core/apps/authentication-portal/src/main/webapp/fido2-error.jsp:38]: We can keep this as a boolean, can't we?

File: identity-apps-core/apps/authentication-portal/src/main/webapp/fido2-error.jsp [hunk 5]
```diff
@@ -0,0 +1,135 @@
...
+                        <div class="ui divider hidden"></div>
+                    <% } %>
+                </div>
+                <div class="ui bottom attached warning message">
+                    <p class="text-left mt-0">
+                        <%=AuthenticationEndpointUtil.i18n(resourceBundle, "need.help.contact.us")%>
+                        <a href="mailto:<%= StringEscapeUtils.escapeHtml4(supportEmail) %>" target="_blank">
+                            <span class="orange-text-color button"><%= StringEscapeUtils.escapeHtml4(supportEmail) %></span>
+                        </a> 
+
```
Comment (pavinduLakshan) [inline identity-apps-core/apps/authentication-portal/src/main/webapp/fido2-error.jsp:104]:
```suggestion
```

File: identity-apps-core/apps/authentication-portal/src/main/webapp/includes/product-footer.jsp [hunk 6]
```diff
@@ -130,7 +130,7 @@
                 List<String> langSwitcherEnabledServlets = Arrays.asList("/oauth2_login.do", "/oauth2_error.do",
                     "/confirmregistration.do", "/confirmrecovery.do", "/claims.do", "/oauth2_consent.do",
                     "/fido2-auth.jsp", "/fido2-identifierfirst.jsp", "/fido2-enroll.jsp", "/fido2-passkey-status.jsp",
-                    "/email_otp.do", "/org_name.do", "/org_discovery.do", "/retry.do", "/totp_enroll.do",
+                    "/fido2-error.jsp", "/email_otp.do", "/org_name.do", "/org_discovery.do", "/retry.do", "/totp_enroll.do",
```
Comment (pavinduLakshan) [inline identity-apps-core/apps/authentication-portal/src/main/webapp/includes/product-footer.jsp:133]: Please check whether the content in fido2-error.jsp persists between browser refreshes. If it doesn't, let's not enable lang switcher for this page.

Comment (AfraHussaindeen) [reply identity-apps-core/apps/authentication-portal/src/main/webapp/includes/product-footer.jsp:133]: The error message persist even after refresh

File: identity-apps-core/apps/authentication-portal/src/main/webapp/fido2-error.jsp (hunk 5 above)
Comment (AfraHussaindeen) [reply identity-apps-core/apps/authentication-portal/src/main/webapp/fido2-error.jsp]: Resolved via https://github.com/wso2/identity-apps/pull/4475/commits/14ac6fedb995950ed179be3be2086d78733ce378

File: identity-apps-core/apps/authentication-portal/src/main/webapp/fido2-error.jsp (hunk 4 above)
Comment (AfraHussaindeen) [reply identity-apps-core/apps/authentication-portal/src/main/webapp/fido2-error.jsp]: Resolved via https://github.com/wso2/identity-apps/pull/4475/commits/14ac6fedb995950ed179be3be2086d78733ce378

File: identity-apps-core/apps/authentication-portal/src/main/webapp/fido2-error.jsp (hunk 3 above)
Comment (AfraHussaindeen) [reply identity-apps-core/apps/authentication-portal/src/main/webapp/fido2-error.jsp]: Resolved via  https://github.com/wso2/identity-apps/pull/4475/commits/14ac6fedb995950ed179be3be2086d78733ce378

File: identity-apps-core/apps/authentication-portal/src/main/webapp/fido2-error.jsp (hunk 2 above)
Comment (AfraHussaindeen) [reply identity-apps-core/apps/authentication-portal/src/main/webapp/fido2-error.jsp]: Resolved via https://github.com/wso2/identity-apps/pull/4475/commits/14ac6fedb995950ed179be3be2086d78733ce378

File: identity-apps-core/apps/authentication-portal/src/main/webapp/fido2-error.jsp (hunk 1 above)
Comment (AfraHussaindeen) [reply identity-apps-core/apps/authentication-portal/src/main/webapp/fido2-error.jsp:78]: Resolved via https://github.com/wso2/identity-apps/pull/4475/commits/14ac6fedb995950ed179be3be2086d78733ce378

File: identity-apps-core/apps/authentication-portal/src/main/webapp/fido2-error.jsp [hunk 7]
```diff
@@ -0,0 +1,135 @@
...
+  ~ specific language governing permissions and limitations
+  ~ under the License.
+--%>
+
+<%@ page import="org.apache.commons.text.StringEscapeUtils" %>
+<%@ page import="org.owasp.encoder.Encode" %>
+<%@ page import="org.wso2.carbon.identity.application.authentication.endpoint.util.Constants" %>
+<%@ page import="java.io.File" %>
+<%@ page language="java" contentType="text/html; charset=UTF-8" pageEncoding="UTF-8" %>
+<%@ taglib prefix="fmt" uri="http://java.sun.com/jsp/jstl/fmt" %>
```
[the remark this answers is not in the document]
Comment (AfraHussaindeen) [reply identity-apps-core/apps/authentication-portal/src/main/webapp/fido2-error.jsp]: Resolved via https://github.com/wso2/identity-apps/pull/4475/commits/14ac6fedb995950ed179be3be2086d78733ce378

**Review by pavinduLakshan — APPROVED** [review]

**Pull request merged**